src/StoreApi/Routes/Checkout.php

```diff
@@ -277,6 +282,19 @@ protected function get_route_error_response( $error_code, $error_message, $http_
 				$controller = new CartController();
 				$cart       = $controller->get_cart_instance();
 
+				if ( is_wp_error( $error_message ) ) {
```

Comment:
Thanks for the suggestion, I've now added `add_data_to_error_object` to `Checkout` I don't think it needs to be defined in `AbstractRoute` because it's not used anywhere else except the Checkout. If other routes start returning `WP_Error` objects then we can move it but I think that's out of scope for this PR.